plugins/net_plugin/net_plugin.cpp

```diff
@@ -2968,6 +2971,7 @@ namespace eosio {
             peer_ilog( this, "received block ${n} less than ${which}lib ${lib}",
                        ("n", blk_num)("which", blk_num < last_sent_lib ? "sent " : "")
                        ("lib", blk_num < last_sent_lib ? last_sent_lib : lib) );
+            my_impl->sync_master->reset_last_requested_num();
```

Comment:
`sync_mtx` needs acquired to set `sync_last_requested_num`.

Reply:
done and added another reset as i found another test case where we are getting unlinkable block, sending handshake and after lib notice while starting syncing nodeos just logs something like:
`ignoring request, head is 641 last req = 696, source connection 2`

Reply:
@heifner stability test (except mac) with 500 runs passed:
https://buildkite.com/EOSIO/eosio-test-stability/builds/343